pepys_import/core/validators/enhanced_validator.py

```diff
@@ -169,7 +169,9 @@ def speed_loose_match_with_location(
         """
         distance = distance_between_two_points_haversine(prev_location, curr_location)
         calculated_speed = distance / time
-        if speed is None or calculated_speed <= speed * 10:
+        if speed is None:
+            return True
+        elif (speed / 10) <= calculated_speed <= (speed * 10):
```

Comment:
wow @robintw - I didn't think it would be possible to represent the logic in this way.

Reply:
Yeah, Python is cool 😄